specs/Ihec_metadata_specification.md

```diff
@@ -191,6 +191,15 @@ __DONOR_ETHNICITY__ - The ethnicity of the donor that provided the primary tissu
 
 # EXPERIMENTS
 
+###### Note for metadata resubmission
+In order to pass IHEC metadata validation, all datasets submitted prior to 2018 __must__ include the following properties:
+
+* LIBRARY_STRATEGY
+* EXPERIMENT_TYPE or EXPERIMENT_ONTOLOGY_URI
+* MOLECULE or MOLECULE_ONTOLOGY_URI, either defined in the experiment or sample object.
```

Comment:
It's important for instance in the case of RNA-Seq, where you may want to make a distinction between polyA RNA and total RNA.
For verification, didn't we say we'd leave implementation details out of the spec?

Reply:
The point here is to express the changes that were voted for in Dec. 2017. Moving the MOLECULE field to EXPERIMENT is one of these changes. I can add a note that mentions that old submissions can have MOLECULE defined in the SAMPLE object.

Reply:
The IHEC spec does not have a mechanism for cross referencing. The EXPERIMENT for all reasons can contain an EGA accession for the SAMPLE_REF, and the SAMPLE xml may not have its accession. Which makes this not impossible to implement. That's not a question of implementation details. 

As per Dec 2017 mods, MOLECULE goes into EXPERIMENT for 1.0 which is fine, for "fundamental" spec, picking anything other than SAMPLE, can either not be validated, or will cause old submissions to fail validation. The point of "fundamental" spec is to support what people *should* have been submitting, as per which MOLECULE goes in SAMPLE. 

Either way this discussion only effects RNA-Seq. So can we move MOLECULE discussion to only RNA-Seq and not all experiments? Unless there's another reason.

Reply:
Hi Sita, sorry but I'm not sure I understand what's the issue with the current sentence. It say:
"MOLECULE or MOLECULE_ONTOLOGY_URI, either defined in the experiment or sample object." I thought it's clear that way, you can either define a MOLECULE field in the EXPERIMENT object, or in the SAMPLE object.

Reply:
If you want to force a MOLECULE field in SAMPLE for old datasets, you're basically requesting many groups to create a whole bunch of new SAMPLE objects for any missing library samples, and to rework the wiring to experiments. The agreement was that we'll support both ways for old datasets, either in SAMPLE or EXPERIMENT.

Reply:
It cannot be validated. 

Suppose MOLECULE (or MOLECULE_ONTOLOGY_URI) is not in EXPERIMENT. And suppose it is not in SAMPLE either. Then at least one of EXPERIMENT and SAMPLE is invalid. So the one of them must fail validation. The simplest way to do this is if SAMPLE validates without MOLECULE then MOLECULE becomes required for EXPERIMENT.  

But this conditional validation involves cross referencing - you start with an EXPERIMENT and look at the SAMPLE it's linked to - but there's no guarantee that a unique key tying them together exists (e.g. EGA refname or EGA accession - because different objects may choose different ways to reference each other, one uses accession, other alias; but we don't know alias-accession mapping).

So cross referencing is not always possible within IHEC spec. Which means it cannot be validated that the MOLECULE exists in one of SAMPLE or EXPERIMENT if we allow the flexibility of allowing it in either .

Reply:
I see, didn't take into account the many ways that EGA objects can reference each other. So for validation of old data, I guess it's one of these scenarios?
1- Validate this in EpiRR.
2- Don't validate it for old datasets.
3- Issue a warning if MOLECULE is not in EXPERIMENT. Can be safely ignored if the group knows it's in SAMPLE. And rely on good faith that people want to submit their data properly.

Reply:
I don't think we want to break up validation into multiple places. It just becomes confusing as to who is responsible for what. For a placeholder mechanism, I suppose only for old data, we can off load this to EpiRR (since EpiRR will need to verify validation anyway). But moving forward, there should be one definitive validator.  

For old data, when validating EXPERIMENTS, the version we issue is not "fundamental" but "fundamental-no-molecule" in addition to the warning if no molecule is found. So it's clear that the submitter has this responsibility, and may encounter this at EpiRR.   

Since we want to use the same code at EpiRR as well, it's probably worth it to build in the cross referencing assuming a unique key can be found. So there's no getting around building that. It shouldn't be too bad - we'll just need to pull refnames from the xml if they exists and key on it. 

The problem is that some of this really can't be fixed because people may be using same SAMPLE for ChIP and RNA, which means MOLECULE really has to be in EXPERIMENT.

Reply:
Yes, that sounds like it! So are we OK to merge this branch to master?

Reply:
I am good. Does this discussion need to be reflected in the spec anywhere?

Reply:
Sure, I just added a note in the spec. Myrto, can you confirm that this is good on your side as well? We'll then be ready to merge!